apache-rat-core/src/main/java/org/apache/rat/document/impl/DocumentName.java

```diff
@@ -0,0 +1,220 @@
+    /** The full name for the document */
+    private final String name;
+    /** The name of the base directory for the document */
+    private final String baseName;
+    /** The directory separator for this document. */
+    private final String dirSeparator;
+    /** The case-sensitive flag */
+    private final boolean isCaseSensitive;
+
+    static {
+        boolean fsSensitive = true;
+        try {
+            Path p = Files.createTempDirectory("NameSet");
+            File f = p.toFile();
+            fsSensitive = !new File(f, "a").equals(new File(f, "A"));
+            FileUtils.deleteDirectory(f);
+        } catch (IOException e) {
+            fsSensitive = true;
+        }
+        FS_IS_CASE_SENSITIVE = fsSensitive;
+    }
+
+    /**
+     * Creates a Dcoument name.
+     * @param name the name of the document
+     * @param baseName the base name of the document.
+     * @param dirSeparator the directory separator used in the name.
+     * @param isCaseSensitive {@code true} if the name is case-sensitive.
+     */
+    public DocumentName(final String name, final String baseName, final String dirSeparator, final boolean isCaseSensitive) {
+        this.name = Objects.requireNonNull(name);
+        this.baseName = Objects.requireNonNull(StringUtils.defaultIfEmpty(baseName, null));
+        this.dirSeparator = Objects.requireNonNull(dirSeparator);
+        this.isCaseSensitive = isCaseSensitive;
+    }
+
+    /**
+     * Creates a document name with the name of the file and the same basename as the baseName document.
+     * @param file the file to name the document from.
+     * @param baseName the DocumentName to provide the baseName.
+     */
+    public DocumentName(final File file, final DocumentName baseName) {
+        this(file.getAbsolutePath(), baseName.baseName, File.separator, FS_IS_CASE_SENSITIVE);
+    }
+
+    /**
+     * Creates a document name with the name and basename equal to the file name
+     * @param file the file name to use.
+     */
+    public DocumentName(final File file) {
+        this(file.getAbsolutePath(), file.getAbsolutePath(), File.separator, FS_IS_CASE_SENSITIVE);
+    }
+
+    /**
+     * Creates a new document name by adding the child to the current name.
+     * @param child the child to add (must use directory separator from this document name.
+     * @return the new document name with the same base name, directory separator and case sensitivity as this one.
+     */
+    public DocumentName resolve(final String child) {
+        List<String> parts = new ArrayList<>();
+        parts.addAll(Arrays.asList(tokenize(name)));
+        parts.addAll(Arrays.asList(tokenize(child)));
+        String newName = String.join(dirSeparator, parts);
+        return new DocumentName(newName, baseName, dirSeparator, isCaseSensitive);
+    }
+
+    /**
+     * Gets the fully qualified name of the document.
+     * @return the fully qualified name of the document.
+     */
+    public String name() {
```

Comment:
to be compliant with the Java-rules this should be getName() as the class is not an enum - WDYT?